hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/verifier/DelegateHandleContextVerifier.java

```diff
@@ -0,0 +1,126 @@
+import com.hedera.node.app.spi.workflows.VerificationAssistant;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.function.Predicate;
+
+/**
+ * A {@link HandleContextVerifier} that delegates resolves complex keys and passes checks for primitive keys
+ * to a provided {@link Predicate}-verifier.
+ */
+public class DelegateHandleContextVerifier implements HandleContextVerifier {
+
+    private final Predicate<Key> baseVerifier;
+
+    /**
+     * Constructs a {@link DelegateHandleContextVerifier}
+     *
+     * @param baseVerifier the base verifier
+     */
+    public DelegateHandleContextVerifier(@NonNull final Predicate<Key> baseVerifier) {
+        this.baseVerifier = requireNonNull(baseVerifier, "baseVerifier must not be null");
+    }
+
+    @NonNull
+    @Override
+    public SignatureVerification verificationFor(@NonNull final Key key) {
+        requireNonNull(key, "key must not be null");
+        return doVerification(key, baseVerifier);
+    }
+
+    @NonNull
+    @Override
+    public SignatureVerification verificationFor(
+            @NonNull final Key key, @NonNull final VerificationAssistant callback) {
+        requireNonNull(key, "key must not be null");
+        requireNonNull(callback, "callback must not be null");
+        final Predicate<Key> composedVerifier = key1 -> {
+            final var intermediateVerification = baseVerifier.test(key1)
+                    ? SignatureVerificationImpl.passedVerification(key1)
+                    : SignatureVerificationImpl.failedVerification(key1);
+            return callback.test(key1, intermediateVerification);
+        };
+        return doVerification(key, composedVerifier);
+    }
+
+    @NonNull
+    @Override
+    public SignatureVerification verificationFor(@NonNull Bytes evmAlias) {
+        requireNonNull(evmAlias, "evmAlias must not be null");
+        return failedVerification(evmAlias);
+    }
+
+    @Override
+    public int numSignaturesVerified() {
+        return 0;
+    }
+
+    /**
+     * Does a complex verification of a {@link Key} using the provided {@link Predicate} to check primitive keys.
+     *
+     * @param key the {@link Key} to verify
+     * @param primitiveVerifier the {@link Predicate} to use to verify primitive keys
+     * @return the {@link SignatureVerification} result
+     */
+    @NonNull
+    private static SignatureVerification doVerification(
+            @NonNull final Key key, @NonNull final Predicate<Key> primitiveVerifier) {
+        final var result =
+                switch (key.key().kind()) {
+                    case KEY_LIST -> {
+                        final var keys = key.keyListOrThrow().keysOrElse(emptyList());
+                        for (final var childKey : keys) {
+                            if (doVerification(childKey, primitiveVerifier).failed()) {
+                                yield false;
```

Comment:
When the primitiveVerifier is present, we cannot exit early.  The contract for the verifiers is that they get to observe all of the primitive keys, not just some.  This is critical because schedule is gathering primitive keys, and skipping some during a create or sign operation could prevent later execution of the scheduled transaction when it should succeed.
This mostly applies to lists (where we might only have some, and an early key could fail initially), but also applies to thresholds where an early success could prevent gathering a primitive needed in a later sign transaction.

Reply:
Right… I was wondering why I implemented it like that before. Thanks for reminding me! 😁